### PR TITLE
Vet Clinic #2: Query and update animals table

### PR DESCRIPTION
**Pull request by danifromecuador**
## In this pull request, I made the following tasks:
- Create a `query-and-update-animals-table` branch
- Add a `species` column to my `animals`
  -  `ALTER TABLE animals ADD COLUMN species VARCHAR(255);`
  image
- Insert 7 new pokemos to `animals` table
  -  `INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg, species)
VALUES ('Charmander', '2020-02-08', 0, FALSE, -11, NULL);
INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg, species)
VALUES ('Plantmon', '2021-11-15', 2, TRUE, -5.7, NULL);
INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg, species)
VALUES ('Squirtle', '1993-04-02', 3, FALSE, -12.13, NULL);
INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg, species)
VALUES ('Angemon', '2005-06-12', 1, TRUE, -45, NULL);
INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg, species)
VALUES ('Boarmon', '2005-06-07', 7, TRUE, 20.4, NULL);
INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg, species)
VALUES ('Blossom', '1998-10-13', 3, TRUE, 17, NULL);
INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg, species)
VALUES ('Ditto', '2022-05-14', 4, TRUE, 22, NULL);`
image
- Inside a transaction update the animals table by setting the species column to unspecified
  -  `UPDATE animals SET species = 'unspecified';`
image
- Then roll back the change and verify that the species columns went back to the state before the transaction
image
-  Inside a transaction
    -  Update the animals table by setting the species column to digimon for all animals that have a name ending in mon
    - Update the animals table by setting the species column to pokemon for all animals that don't have species already set
    image
- Inside a transaction delete all records in the animals table, then roll back the transaction.
image
- Inside a transaction:
  - Delete all animals born after Jan 1st, 2022.
  - Create a savepoint for the transaction.
  - Update all animals' weight to be their weight multiplied by -1.
  - Rollback to the savepoint
  - Update all animals' weights that are negative to be their weight multiplied by -1.
  - Commit transaction  
image
- Write queries to answer the following questions:
  - How many animals are there?
  - How many animals have never tried to escape?
  - What is the average weight of animals?
  - Who escapes the most, neutered or not neutered animals?
  - What is the minimum and maximum weight of each type of animal?
  - What is the average number of escape attempts per animal type of those born between 1990 and 2000?  
image

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -7,5 +7,6 @@ vet_clinic=# CREATE TABLE animals (
     date_of_birth DATE,
     escape_attempts INTEGER,
     neutered BOOLEAN,
-    weight_kg DECIMAL(10, 2)
+    weight_kg DECIMAL(10, 2),
+    species VARCHAR(255)
```
Comment (Ibizugbe) [inline schema.sql]: I noticed you added the species column directly in your code😉. But this won't be feasible on your code editor😉. The table has already been created and we just want to add a new column. you did the right thing as shown in the PR description by using the `alter table` method😉. Kindly use that same method here to add the species column to your animal table😉.

you should use this instead after the create table `ALTER TABLE animals ADD COLUMN species VARCHAR(255);` I picked that from your PR description.

File: queries.sql [hunk 2]
```diff
@@ -21,3 +21,59 @@ SELECT * FROM animals WHERE name <> 'Gabumon';
...
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+
+-- Update the animals table by setting the species column to pokemon for all animals that don't have species already set.
+UPDATE animals SET species = 'pokemon' WHERE species IS NULL;
+COMMIT;
+
+--Inside a transaction delete all records in the animals table, then roll back the transaction.
+BEGIN;
+DELETE FROM animals;
+ROLLBACK;
```
Comment (Ibizugbe) [inline queries.sql]: for the requirements to update the table and also to delete some records, I noticed you didn't add the code to verify that the action was successful as required. please kindly add the code to show that the actions you performed worked. you did this properly in your PR description but you didn't do it here. and it is more important here😉.

image

**Review by Ibizugbe — CHANGES_REQUESTED** [review]
Hi @danifromecuador 👋🏼👋🏼👋🏼, 

require_changes


Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights✅✅✅

- [x] PR contains a properly documented description of the actions carried out with screen shots for physical evidence👍🏼.
- [x] The required data has been added to the table👍🏼.


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema.sql (hunk 1 above)
Comment (danifromecuador) [reply schema.sql]: Thank you dear @Ibizugbe Your review is very detailed.

File: queries.sql (hunk 2 above)
Comment (danifromecuador) [reply queries.sql]: Thank you @Ibizugbe I've included `SELECT * FROM animals;` inside each transaction to verify if the changes were made.

File: schema.sql (hunk 1 above)
Comment (rloterh) [reply schema.sql]: well resolved! ✔️

File: queries.sql (hunk 2 above)
Comment (rloterh) [reply queries.sql]: well resolved! ✔️

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Dani,  

Your project is complete! 🏁 There is nothing else to say other than... it's time to merge it 🍾 Congratulations!  🎉💯🌟 

## Highlights 🎯 
🎲 Good job for introducing improvements to update queries!

## Optional suggestions 🧰 
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

**You can also consider:**
The _Getting Started_ aspect of the ReadMe is occurring twice, lets endeavor to consolidate them.
Implementing the above change(s) will make your ReadM stand out even better! 😉

Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**